build_app.sh

```diff
@@ -10,12 +10,23 @@ application="Applications/$1"
 echo "$application"
 
 if [ ! -d $application ]
-then 
+then
     echo "No directory named after given application."
     exit 1
 fi
 
 # Do a backup, just in case
-mv app_src/* app_src_backup
-cp -r $application/. app_src/
-./docker_build_image.sh
\ No newline at end of file
+BACKUP_DIR=app_src_backup
+NEW_DIR_NAME="backup_"$(date +%s)
+mkdir $BACKUP_DIR/$NEW_DIR_NAME
+
+if [ "$(ls -A app_src -I ".gitkeep")" ]
+then
+  mv app_src/* $BACKUP_DIR/$NEW_DIR_NAME
+  cp -r $application/. app_src/
+else
+   cp -r $application/. app_src/
+fi
+
+#SUDO privileges are required
+sudo ./docker_build_image.sh
```

Comment:
Why? You don't and should not need sudo privileges to build application.

Reply:
Docker needs a sudo user to run as described here: https://docs.docker.com/engine/install/linux-postinstall/
If this needs to be avoided the developer need to create a new group and add himeself to that group. 
Is there any specific reason why sudo privileges needs to be avoided?

Reply:
Using sudo in build script complicates the building process in my opinion. Adding docker to new group is needed just once, and then, you can build app with few key strokes. Using sudo inside scripts require sthe user to enter a password each time, which, in my opinion, sometimes might be annoying, especially, when for example, you just miss spelled the variable name and need to build just after the previous build.

Reply:
It is perfectly fine to me given that this is described somewere in the documentations. Not all developers are familiar with linux and hence these details can be ignored raising issues.
What about managing the group creation automatically on the script? Would that be something usefull in your opinion?

Reply:
@Ruben-Crispino yes, it would be useful in my opinion

Reply:
@magiczny-kacper check new commit